Sources/BranchSDK/Branch.m

```diff
@@ -624,27 +625,20 @@ - (void)initSessionWithLaunchOptions:(NSDictionary *)options
     [self.class addBranchSDKVersionToCrashlyticsReport];
     self.shouldAutomaticallyDeepLink = automaticallyDisplayController;
 
-    // If the SDK is already initialized, this means that initSession was called after other lifecycle calls.
-    if (self.initializationStatus == BNCInitStatusInitialized) {
-        [self initUserSessionAndCallCallback:YES sceneIdentifier:nil urlString:nil];
-        return;
-    }
-
-    // Save data from push notification on app launch
+    // Check for Branch link in a push payload
+    NSString *pushURL = nil;
     #if !TARGET_OS_TV
     if ([options objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey]) {
         id branchUrlFromPush = [options objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey][BRANCH_PUSH_NOTIFICATION_PAYLOAD_KEY];
         if ([branchUrlFromPush isKindOfClass:[NSString class]]) {
             self.preferenceHelper.universalLinkUrl = branchUrlFromPush;
             self.preferenceHelper.referringURL = branchUrlFromPush;
+            pushURL = (NSString *)branchUrlFromPush;
         }
     }
     #endif
 
-    // Handle case where there's no URI scheme or Universal Link.
-    if (![options.allKeys containsObject:UIApplicationLaunchOptionsURLKey] && ![options.allKeys containsObject:UIApplicationLaunchOptionsUserActivityDictionaryKey]) {
-        [self initUserSessionAndCallCallback:YES sceneIdentifier:nil urlString:nil];
-    }
+    [self initUserSessionAndCallCallback:YES sceneIdentifier:nil urlString:pushURL];
```

Comment:
Can you add a comment if there are any assumptions about from which lifecycle call(s) this must go through?

Reply:
Previously we assumed that initSession always came first, and other lifecycle calls would follow. With the delay feature, we cannot assume that anymore and cache away links arriving from other lifecycle calls. So now there are no assumptions about where the initSession is coming from.